src/VehicleSetup/JoystickConfigButtons.qml

```diff
@@ -163,24 +165,85 @@ ColumnLayout {
                     }
                 }
 
-                FactComboBox {
-                    id:             mainJSButtonActionCombo
-                    width:          ScreenTools.defaultFontPixelWidth * 15
-                    fact:           hasFirmwareSupport ? controller.getParameterFact(-1, parameterName) : null;
-                    visible:        hasFirmwareSupport
-                    indexModel:     false
+                QGCComboBox {
+                    id:                         buttonActionCombo
+                    width:                      ScreenTools.defaultFontPixelWidth * 26
+                    property Fact fact:         controller.parameterExists(-1, `BTN${index}_FUNCTION`) ? controller.getParameterFact(-1, `BTN${index}_FUNCTION`) : null
+                    property Fact fact_shift:   controller.parameterExists(-1, `BTN${index}_SFUNCTION`) ? controller.getParameterFact(-1, `BTN${index}_SFUNCTION`) : null
+                    property var factOptions:   fact ? fact.enumStrings : [];
+                    property var qgcActions:    _activeJoystick.assignableActionTitles.filter(function(s){return s.includes("Camera") || s.includes("Stream") || s.includes("Zoom") || s.includes("Gimbal")  || s.includes("No Action")})
+                    model:                      [...qgcActions, ...factOptions]
+                    property var isFwAction:    currentIndex >= qgcActions.length
                     sizeToContents: true
+
+                    function _findCurrentButtonAction() {
+                        // Find the index in the dropdown of the current action, checks FW and QGC actions
+                        if(_activeJoystick) {
+                            if (fact && fact.value > 0) {
+                                // This is a firmware function
+                                currentIndex = qgcActions.length + fact.value
+                                // For sanity reasons, make sure qgc is set to "no action" if the firmware is set to do something
+                                _activeJoystick.setButtonAction(modelData, "No Action")
+                            } else {
+                                // If there is not firmware function, check QGC ones
+                                var i = find(_activeJoystick.buttonActions[modelData])
```

Comment:
this intermediary variable is a bit unnecessary, and currentIndex already provides the information that find will return an index like variable.